### PR TITLE
feat(form): Refactor some small components to separate modules

### PR DESCRIPTION
**Pull request by billyvg**
This moves some error state/control state components into a module and allow normal `<Field>` components to have an error and control state (e.g. loading, saved, and error states)

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx [hunk 1]
```diff
@@ -91,71 +30,6 @@ const getValueFromEvent = (valueOrEvent, e) => {
...
 };
 
-const ControlStateWrapper = styled('div')`
-  padding: 0 8px;
-`;
-
-/**
- * ControlState (i.e. loading/error icons) for connected form components
- */
-class ControlState extends React.Component {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx]: Moved and split between field and formField where the latter uses the model.

File: src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx [hunk 2]
```diff
@@ -4,78 +4,17 @@ import React from 'react';
...
-  display: flex;
-  align-items: center;
-  justify-content: center;
-`;
-
-const FormSpinner = styled(Spinner)`
-  margin-left: 0;
-`;
-
-const ReturnButtonStyled = styled(returnButton)`
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx]: Moved to bottom of file

File: src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx [hunk 3]
```diff
@@ -4,78 +4,17 @@ import React from 'react';
...
+import FieldErrorReason from 'app/views/settings/components/forms/field/fieldErrorReason';
+import FormFieldControlState from 'app/views/settings/components/forms/formField/controlState';
 import PanelAlert from 'app/components/panels/panelAlert';
-import Spinner from 'app/views/settings/components/forms/spinner';
-import returnButton from 'app/views/settings/components/forms/returnButton';
-import {sanitizeQuerySelector} from 'app/utils/sanitizeQuerySelector';
+import ReturnButton from 'app/views/settings/components/forms/returnButton';
 import space from 'app/styles/space';
 
-const FormFieldErrorReason = styled('div')`
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx]: Moved into separate module

File: src/sentry/static/sentry/app/views/settings/components/forms/field/fieldErrorReason.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+const FieldErrorReason = styled('div')`
+  color: ${p => p.theme.redDark};
+  position: absolute;
+  right: 2px;
+  margin-top: 6px;
+  background: #fff;
+  padding: 6px 8px;
+  font-weight: 600;
+  font-size: 12px;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/components/forms/field/fieldErrorReason.tsx]: Should we use a theme prop for the font-size?

File: src/sentry/static/sentry/app/views/settings/components/forms/field/fieldErrorReason.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  border-radius: 3px;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/components/forms/field/fieldErrorReason.tsx]: This isn't one of our normal border-radius sizes. Should it use theme?

Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/components/forms/field/fieldErrorReason.tsx]: I'd prefer not to change design right now

**Review by dashed — APPROVED** [review]
Looks good to me 👍

**Pull request merged**